### ISSUE TITLE
IndentationCheck false positive with double-brace initialization

### ISSUE THREAD

**Issue by Carbon14-48** [checkstyle/checkstyle#19415]
I have read check documentation: https://checkstyle.org/checks/misc/indentation.html  
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run  
I have executed the CLI and showed it below, as CLI describes the problem better than 1,000 words.

```bash
/var/tmp $ javac HashMapTestDefault.java
Successfully compiled with JDK 21

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation"/>
    </module>
</module>

/var/tmp $ cat HashMapTestDefault.java
import java.util.HashMap;

public class HashMapTestDefault {
    @Test
    void test() {
        var map = new HashMap<>() {{
            put("KEY1", "VALUE1");
            put("KEY2", "VALUE2");
        }};
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.3.0-all.jar -c config.xml HashMapTestDefault.java
Starting audit...
[ERROR] /var/tmp/HashMapTestDefault.java:7:13: 'block' child has incorrect indentation level 12, expected level should be one of the following: 16, 20. [Indentation]
[ERROR] /var/tmp/HashMapTestDefault.java:8:13: 'block' child has incorrect indentation level 12, expected level should be one of the following: 16, 20. [Indentation]
[ERROR] /var/tmp/HashMapTestDefault.java:9:9: 'block rcurly' has incorrect indentation level 8, expected level should be one of the following: 12, 16. [Indentation]
Audit done.
Checkstyle ends with 3 errors.
```

---

### Describe what you expect in detail

No violations should be reported. The code follows standard 4-space indentation:

- 4 spaces for method/annotation
- 8 spaces for variable declaration
- 12 spaces for statements inside instance initializer
- 8 spaces for closing braces

The double-brace initialization pattern `new HashMap<>() {{ ... }}` creates an anonymous class with an instance initializer. The indentation check incorrectly calculates the expected indentation for statements inside the instance initializer and for the closing braces. 
 _Originally posted by @SButterfly in [#2829](https://github.com/checkstyle/checkstyle/issues/2829#issuecomment-1387174388)_

**Comment by Carbon14-48** [comment]
@romani  if this gets approved, I'd like to work on it. I already have experience with the IndentationCheck from #19360  (nested method calls fix), which is waiting for review. This double-brace initialization issue seems similar in nature.

Please let me know if that's okay!

**Comment by romani** [comment]
some attempt was made at https://github.com/checkstyle/checkstyle/pull/19471/changes/11cba5daee108dc1041bd1da75bb9f60b783d866 , but abandoned.

Some analysys is done and Test to quickly reproduce is merged https://github.com/checkstyle/checkstyle/pull/19471

**Comment by romani** [comment]
test to reproduce problems is merged at https://github.com/checkstyle/checkstyle/pull/19471, everyone is welcome to fix issue.

**Comment by kalayciburak** [comment]
I am on it.
